examples/ssd1306/src/main.rs

```diff
@@ -0,0 +1,165 @@
+#![no_main]
+#![no_std]
+#![feature(default_alloc_error_handler)]
+
+extern crate alloc;
+
+use alloc::vec;
+use alloc_cortex_m::CortexMHeap;
+use cherry::widget::{
+    container::{Alignment, Axis, Border, Container, Insets, Justification, Options},
+    Widget,
+};
+use defmt_rtt as _;
+use embedded_graphics::{pixelcolor::BinaryColor, prelude::*, primitives::CornerRadii};
+use panic_probe as _;
+use ssd1306::{mode::BufferedGraphicsMode, prelude::*, I2CDisplayInterface, Ssd1306};
+use stm32f3_discovery::stm32f3xx_hal::{
+    self as _,
+    gpio::{Alternate, Gpiob, OpenDrain, Pin, U},
+    i2c::I2c,
+    pac::{self, I2C1},
+    prelude::*,
+};
+
+type Screen = Ssd1306<
+    I2CInterface<
+        I2c<
+            I2C1,
+            (
+                Pin<Gpiob, U<6_u8>, Alternate<OpenDrain, 4_u8>>,
+                Pin<Gpiob, U<7_u8>, Alternate<OpenDrain, 4_u8>>,
+            ),
+        >,
+    >,
+    ssd1306::prelude::DisplaySize128x64,
+    BufferedGraphicsMode<ssd1306::prelude::DisplaySize128x64>,
+>;
+
+const HEAP_SIZE: usize = 1024;
+
+const MAX_BLOCK_SIZE: u32 = 30;
+const MIN_BLOCK_SIZE: u32 = 0;
+const BLOCK_SIZE_INCREMENT: u32 = 2;
+
+#[global_allocator]
+static ALLOCATOR: CortexMHeap = CortexMHeap::empty();
+
+#[cortex_m_rt::entry]
+fn main() -> ! {
+    init_allocator();
+
+    let mut display = configure_display();
+    animate(&mut display);
+}
+
+fn init_allocator() {
+    static mut HEAP: [u8; HEAP_SIZE] = [0; HEAP_SIZE];
+
+    unsafe {
+        ALLOCATOR.init(
+            &mut HEAP as *const u8 as usize,
+            core::mem::size_of_val(&HEAP),
+        )
```

Comment:
As I understand it, this allocates a chunk of free space in the stack, and then using that space as the heap. That avoids the issue where the stack and heap can collide, and since the allocator I'm using has a fixed size anyway, I don't think there's really a downside to this. See details here: https://github.com/knurling-rs/flip-link/issues/62